### PR TITLE
[SYSTEMML-621] Remove redundant nested error messages

### PR DESCRIPTION
**Pull request by deroneriksson**
Addresses some redundant nested error messages for read, rand, and matrix statements.

Given the following DML:
```
a=read();
b=matrix();
c=rand(2,2);
```

This previously generated the following output:
```
ERROR a.dml line 1:0 unable to process builtin function expression read:ERROR: a.dml -- line 1, column 0 -- read method must have at least filename parameter
ERROR a.dml line 2:0 unable to process builtin function expression matrix:ERROR: a.dml -- line 2, column 0 -- for matrix statement, must specify at least 3 arguments (in order): data, rows, cols
ERROR a.dml line 3:0 unable to process builtin function expression rand:ERROR: a.dml -- line 3, column 0 -- for Rand Statement all arguments must be named parameters

--------------------------------------------------------------
The following 3 parse issues were encountered:
#1 a.dml [line 1:0] [Validation error] -> a=read();
   unable to process builtin function expression read:ERROR: a.dml -- line 1, column 0 -- read method must have at least filename parameter
#2 a.dml [line 2:0] [Validation error] -> b=matrix();
   unable to process builtin function expression matrix:ERROR: a.dml -- line 2, column 0 -- for matrix statement, must specify at least 3 arguments (in order): data, rows, cols
#3 a.dml [line 3:0] [Validation error] -> c=rand(2,2);
   unable to process builtin function expression rand:ERROR: a.dml -- line 3, column 0 -- for Rand Statement all arguments must be named parameters
--------------------------------------------------------------
```

With this PR it generates:
```
ERROR a.dml line 1:0 read method must have at least filename parameter
ERROR a.dml line 2:0 for matrix statement, must specify at least 3 arguments: data, rows, cols
ERROR a.dml line 3:0 for rand statement, all arguments must be named parameters

--------------------------------------------------------------
The following 3 parse issues were encountered:
#1 a.dml [line 1:0] [Validation error] -> a=read();
   read method must have at least filename parameter
#2 a.dml [line 2:0] [Validation error] -> b=matrix();
   for matrix statement, must specify at least 3 arguments: data, rows, cols
#3 a.dml [line 3:0] [Validation error] -> c=rand(2,2);
   for rand statement, all arguments must be named parameters
--------------------------------------------------------------
```

### CODE REVIEW

File: src/main/java/org/apache/sysml/parser/DataExpression.java [hunk 1]
```diff
@@ -246,31 +257,33 @@ else if (functionName.equals("matrix")){
...
+				if (unnamedParamCount > 5) {
+					errorListener.validationError(blp, bcp, "for matrix statement, at most 5 arguments supported: data, rows, cols, byrow, dimname");
+					return null;
+				}
 				
 			} else {
 				// handle first parameter, which is data and may be unnamed
 				ParameterExpression firstParam = passedParamExprs.get(0);
 				if (firstParam.getName() != null && !firstParam.getName().equals(DataExpression.RAND_DATA)){
 					// throw exception -- must be filename as first parameter
```
Comment (gweidner) [inline src/main/java/org/apache/sysml/parser/DataExpression.java]: Hi Deron - should this 'throw exception' comment also be removed similar to change made in line 278?  Thanks, Glenn

Comment (deroneriksson) [inline src/main/java/org/apache/sysml/parser/DataExpression.java]: Good eye! Yes the comment should be removed too. I'll remove it.

**Comment by deroneriksson** [comment]
Obsolete comment removed.

**Comment by gweidner** [comment]
Thank you @deroneriksson for simplifying the error messages!

LGTM

**Comment by deroneriksson** [comment]
@gweidner Thanks for reviewing! I'll merge.

**Pull request closed without merging**